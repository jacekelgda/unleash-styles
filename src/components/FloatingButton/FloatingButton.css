.RoundRedButton {
  position: relative;
  bottom: 18px;
  right: 18px;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #f63954;
  border: none;
  color: #fff;
  font-size: 20px;
  text-align: center;
  cursor: pointer;
  z-index: 1000;
}

.FloatingButton {
  position: fixed;
}

@media screen and (min-width: 768px) {
	.RoundRedButton {
    bottom: 30px;
    right: 30px;
	}
}
