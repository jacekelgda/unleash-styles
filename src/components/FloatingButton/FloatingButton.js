import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from './FloatingButton.css'

/**
 * Floating Button Component
 */
class FloatingButton extends Component {
  render () {
    const { label, onClick, isFloating } = this.props
    const buttonClasses = classNames(
      styles.RoundRedButton,
      (isFloating) && styles.FloatingButton
    )

    return (
      <button className={buttonClasses} onClick={onClick}>{label}</button>
    )
  }
}

FloatingButton.propTypes = {
  /** Label for button */
  label: PropTypes.string.isRequired,
  /** Function to be triggered after clicked */
  onClick: PropTypes.func.isRequired,
  /** Location of button */
  isFloating: PropTypes.bool.isRequired
}

export default FloatingButton
