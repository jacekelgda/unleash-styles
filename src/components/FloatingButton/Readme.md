Floating Button

    /** Change isFloating to {true} to see floating style */

    <FloatingButton label="+" onClick={() => {}} isFloating={false} />
