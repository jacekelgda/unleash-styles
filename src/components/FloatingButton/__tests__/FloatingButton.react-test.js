/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'

import FloatingButton from '../FloatingButton'

test('FloatingButton renders correctly', () => {
  const wrapper = shallow(<FloatingButton label='+' onClick={() => {}} isFloating />)
  expect(wrapper).toMatchSnapshot()
})
