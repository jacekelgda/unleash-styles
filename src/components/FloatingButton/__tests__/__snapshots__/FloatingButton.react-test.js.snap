// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingButton renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="RoundRedButton FloatingButton"
    onClick={[Function]}
>
    +
</button>,
  "nodes": Array [
    <button
      className="RoundRedButton FloatingButton"
      onClick={[Function]}
>
      +
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FloatingButton
        isFloating={true}
        label="+"
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FloatingButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isFloating": true,
          "label": "+",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="RoundRedButton FloatingButton"
          onClick={[Function]}
>
          +
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="RoundRedButton FloatingButton"
          onClick={[Function]}
>
          +
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FloatingButton
    isFloating={true}
    label="+"
    onClick={[Function]}
/>,
}
`;
